File: udp-server/src/main/java/io/sparkled/udpserver/impl/command/GetVersionCommand.java
package io.sparkled.udpserver.impl.command;

import io.sparkled.model.setting.SettingsCache;
import io.sparkled.music.PlaybackState;

/**
 * Retrieves the current UDP protocol version. This can be used by clients to determine whether they can support the
 * protocol of the Sparkled instance they are talking to.
 * Command syntax: GV
 */
public class GetVersionCommand extends RequestCommand {

    public static final String KEY = "GV";
    private static final int UDP_PROTOCOL_VERSION = 1;

    @Override
    public byte[] getResponse(String[] args, SettingsCache settings, PlaybackState playbackState) {
        return new byte[]{UDP_PROTOCOL_VERSION};
    }
}

File: udp-server/src/main/kotlin/io/sparkled/udpserver/impl/command/GetVersionCommand.kt
package io.sparkled.udpserver.impl.command

import io.sparkled.model.setting.SettingsCacheEntry
import io.sparkled.music.PlaybackState
import java.net.InetAddress

/**
 * Retrieves the current UDP protocol version. This can be used by clients to determine whether they can support the
 * protocol of the Sparkled instance they are talking to.
 * Command syntax: GV
 */
class GetVersionCommand : UdpCommand {

    override fun handle(
        ipAddress: InetAddress,
        port: Int,
        args: List<String>,
        settings: SettingsCacheEntry,
        playbackState: PlaybackState
    ): ByteArray {
        return byteArrayOf(UDP_PROTOCOL_VERSION.toByte())
    }

    companion object {
        const val KEY = "GV"
        private const val UDP_PROTOCOL_VERSION = 1
    }
}
